fibers.js

```diff
@@ -0,0 +1,25 @@
+const Future = require('fibers/future');
+const fs = Future.wrap(require('fs'));
+const prependDir = (file) => require('path').join(process.argv[2], file)
+
+const main = () => {
+  Future.task(function() {
+    try {
+      process.stdout.write(
+        fs
+        .readFileFuture(prependDir('index.txt'))
+        .wait()
+        .toString()
+        .replace(/\n$/,'')
+        .split('\n')
+        .map((x) => fs.readFileFuture(prependDir(x)).wait().toString())
+        .join(''))
```

Comment:
I'm not that great with pureness: read a lot, caught a bit. I'll try
I just use example from [node-fibers](https://github.com/laverdet/node-fibers) 
And I don't think about this Future as a Haskel Future